packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -107,4 +115,83 @@ export class InsightsService {
 			}
 		});
 	}
+
+	// available
+	async getInsightsSummary(): Promise<InsightsSummary> {
+		const rows = await this.insightsByPeriodRepository.getPreviousAndCurrentPeriodTypeAggregates();
+
+		// Initialize data structures for both periods
+		const data = {
+			current: { byType: {} as Record<TypeUnits, number> },
+			previous: { byType: {} as Record<TypeUnits, number> },
+		};
+
+		// Organize data by period and type
+		rows.forEach((row) => {
+			const { period, type, total_value } = row;
+			if (!data[period]) return;
```

Comment:
Should this return something?
An empty object or a default object with 0 values for all keys?

Feels like this might be error-prone if we're return the value via the API (unless this is caught further up the callstack).

Reply:
Yes, and also the `// eslint-disable-next-line @typescript-eslint/no-unsafe-return` comment should be removed as well from the controller